quint/src/parsing/quintParserFrontend.ts

```diff
@@ -116,8 +117,17 @@ export function parsePhase1fromText(
   // run the parser
   const tree = parser.modules()
 
-  // walk through the AST and construct the IR
-  ParseTreeWalker.DEFAULT.walk(listener as QuintListener, tree)
+  // wrap this in a try catch, since we are running the parser even if there are
+  // errors after the previous command.
+  try {
+    // walk through the AST and construct the IR
+    ParseTreeWalker.DEFAULT.walk(listener as QuintListener, tree)
+  } catch (e) {
+    if (errors.length === 0) {
+      throw e
+    }
+    // ignore the exception, we already have errors to report
```

Comment:
But what if the exception is not obviously explained by the recorded errors? Then we would end up swallowing unknown exceptions, potentially hiding errors.

Instead, I suggest this catch only apply to errors which are from a known family of errors that we are sure will be reflected in the `errors`, and anything else is left to bubble up. In my experience, a catch-all that recovers without a record of the exception inevitably comes back to bite you :D

Reply:
> But what if the exception is not obviously explained by the recorded errors?

It is not, what is happening is that the generated code breaks internally when the AST is malformed (because of parser errors). The idea is: there is something being reported, you should fix that problem first and then try again. If an exception is thrown after that (which should be impossible), then you can worry about the exception.

It should always be the case that the exception is a direct consequence of one of the errors, and that fixing the errors will make the exception go away.

> Instead, I suggest this catch only apply to errors which are from a known family of errors that we are sure will be reflected in the `errors`

That I cannot do, because I'm introducing this try catch to handle unpredictable (or hard to predict) errors that are internal to the generated code from antlr4ts. Took me some manual testing to find the first case where some error is thrown, and I don't think I could ever find all possible errors that can be thrown.

Reply:
I'm not following, maybe we can talk about this a bit tomorrow.

Reply:
Thanks for talking thru the reasoning here. I am still a bit worried this may hide problems at some point, but less so and not enough to block here.

Perhaps the comment could be expanded to explain the reasoning?